### ISSUE TITLE
DataView does not have correct toStringTag

### ISSUE THREAD

**Issue by jdalton** [Microsoft/ChakraCore#656]
In Edge 14.14295 a data view has the incorrect [toStringTag](http://www.ecma-international.org/ecma-262/6.0/#sec-dataview.prototype-@@tostringtag).
```js
Object.prototype.toString.call(new DataView(new ArrayBuffer(23)));
// => [object Object] instead of [object DataView]
```

**Comment by nicolo-ribaudo** [comment]
The same is for promises:
```js
Object.prototype.toString.call(new Promise(() => {})); // => [object Object]
```

Note that if you run ch.exe with the `-ES6ToStringTag` flag both the examples work.

**Comment by obastemur** [comment]
Update: #657 handles the both. (`DataView` and `Promise`)

**Comment by ianwjhalliday** [comment]
Fyi `-ES6ToStringTag` turns on the incomplete implementation of the toStringTag feature.  So fixing this the way @obastemur has done is a good thing to do before toStringTag is completed.

**Comment by ianwjhalliday** [comment]
@abchatra and I decided this is a low risk bug fix so we'll include it in 1.2.
